demo/demo/tests/test_subscription_queryset.py

```diff
@@ -0,0 +1,123 @@
+import enum
+from datetime import datetime, timezone
+
+import pytest
+from dateutil.relativedelta import relativedelta
+
+from subscriptions.models import Subscription
+
+
+class SubType(enum.Enum):
+    PAST_ENDING_BEFORE_TRIAL_END = enum.auto()
+    PAST_ENDING_AFTER_TRIAL_END = enum.auto()
+    PAST_WITHOUT_TRIAL = enum.auto()
+    CURRENT_BEFORE_TRIAL_END = enum.auto()
+    CURRENT_AFTER_TRIAL_END = enum.auto()
+    CURRENT_WITHOUT_TRIAL = enum.auto()
+
+
+@pytest.fixture
+def now_() -> datetime:
+    return datetime(2024, 5, 20, tzinfo=timezone.utc)
+
+
+@pytest.fixture
+def subscriptions(plan, user, now_) -> dict[SubType, Subscription]:
+    return {
+        SubType.PAST_ENDING_BEFORE_TRIAL_END: Subscription.objects.create(
+            user=user,
+            plan=plan,
+            auto_prolong=False,
+            initial_charge_offset=relativedelta(days=5),
+            # Two days long subscription, trial period.
+            start=now_ - relativedelta(months=10, days=10),
+            end=now_ - relativedelta(months=10, days=8),
+        ),
```

Comment:
Why is it so complicated? I would just create many fixtures, like this:
```python
@pytest.fixture
def past_ending_before_trial_end_subscription(...) -> Subscription:
    return Subscription.objects.create(...)
```
and then
```python
assert set(charged_subscriptions) == {
    past_ending_before_trial_end_subscription,
    past_without_trial_subscription,
    ...
}
```